examples/ios/objc/Draw/Views/DrawView.m

```diff
@@ -0,0 +1,183 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2014 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "DrawView.h"
+#import "DrawPath.h"
+#import "SwatchesView.h"
+#import "CanvasView.h"
+#import "UIColor+Realm.h"
+#import <Realm/Realm.h>
+
+@interface DrawView ()
+
+@property DrawPath *drawPath;
+@property NSString *pathID;
+@property RLMResults *paths;
+@property RLMNotificationToken *notificationToken;
+@property CanvasView *canvasView;
+@property SwatchesView *swatchesView;
+@property NSString *currentColorName;
+
+@end
+
+@implementation DrawView
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        typeof(self) __weak weakSelf = self;
+        self.notificationToken = [[RLMRealm defaultRealm] addNotificationBlock:^(NSString *notification, RLMRealm *realm) {
+        
+            if (weakSelf.paths.count == 0) {
+                [weakSelf.canvasView clearCanvas];
+            }
+            else {
+                [weakSelf.canvasView setNeedsDisplay];
+            }
+        }];
+        self.paths = [DrawPath allObjects];
+        
+        self.canvasView = [[CanvasView alloc] init];
+        self.canvasView.paths = self.paths;
+        [self addSubview:self.canvasView];
+        
+        self.swatchesView = [[SwatchesView alloc] initWithFrame:CGRectZero];
+        [self addSubview:self.swatchesView];
+        
+        self.swatchesView.swatchColorChangedHandler = ^{
+            weakSelf.currentColorName = weakSelf.swatchesView.selectedColor;
+        };
+        
+        self.currentColorName = @"Black";
+    }
+    return self;
+}
+
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+    
+    CGSize boundsSize = self.bounds.size;
+    CGFloat maxDimension = MAX(boundsSize.width, boundsSize.height);
+    
+    CGRect frame = self.canvasView.frame;
+    frame.size.width = maxDimension;
+    frame.size.height = maxDimension;
+    frame.origin.x = ((boundsSize.width) - maxDimension) * 0.5f;
```

Comment:
can remove one level of parens: `frame.origin.x = (boundsSize.width - maxDimension) * 0.5f;`